tests/sources/generic.py

```diff
@@ -14,28 +14,27 @@
 
 import asyncio
 import random
-from itertools import cycle
 from typing import Any, Dict
 
 
 async def main(queue: asyncio.Queue, args: Dict[str, Any]):
     payload = args.get("payload")
     randomize = args.get("randomize", False)
     delay = int(args.get("delay", 0))
-    loop_mode = args.get("loop_mode", False)
+    loop_count = int(args.get("loop_count", 1))
 
     if not isinstance(payload, list):
         payload = [payload]
 
     if randomize:
         random.shuffle(payload)
 
-    if loop_mode:
-        payload = cycle(payload)
-
-    for event in payload:
-        await queue.put(event)
-        await asyncio.sleep(delay)
+    iteration = 0
+    while iteration != loop_count:
+        for event in payload:
```

Comment:
@Alex-Izquierdo Should we move the random.shuffle in here so its changing for every iteration

Reply:
ok, updated